test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointSmokeTest.java

```diff
@@ -457,6 +462,213 @@ public void checkpointWriterInterleavedTest() throws Exception {
         }
     }
 
+    /** Test the CheckpointWriter write size limit.
+     *
+     * CheckpointWriter aggregates a batch of 50 SMREntries into one
+     * CheckpointEntry. This test uses large-sized SMREntries to verify
+     * that batching will not violate the maxWriteSize limit.
+     */
+    @Test
+    @SuppressWarnings("checkstyle:magicnumber")
+    public void checkpointWriterSizeLimitTest() throws Exception {
+        final String streamName = "mystream5";
+        final UUID streamId = CorfuRuntime.getStreamID(streamName);
+        final String keyPrefix = "a-prefix";
+        final int numKeys = 50;
+        final String author = "Me, myself, and I";
+        final String mutationSuffix = "_mutation_suffix";
+        // max write size set to 25MB
+        final int maxWriteSize = 25 << 20;
+
+        StreamingMap<String, String> m = instantiateStringMap(streamName);
+
+        for (int i = 0; i < numKeys; i++) {
+            // each entry is 1 MB
+            String payload = getRandomStringOfSize(1 << 20);
```

Comment:
Probably need to test both code paths, where compression is enabled and disabled. For the case where compression is enabled, you can either assert that the compressed size is not less than the uncompressed size, or keep generating random strings until that condition is met.

Reply:
I'm not sure if I understand correctly. With the new code which guarantees the uncompressed size of a batch is always smaller than 25 MB, it is impossible to make compressed size > 25 MB in `StreamView::append`. We are testing the old code here since the implementation of `CheckpointWriter::appendObjectState` has changed and it does not blindly combine 50 entries into one. We can't reproduce or test the bug with this PR.